src/Core_Command.php

```diff
@@ -290,28 +296,64 @@ function () use ( $temp ) {
 				'insecure' => $insecure,
 			];
 
-			$response = Utils\http_request( 'GET', $download_url, null, $headers, $options );
+			$retry	     = (int) Utils\get_flag_value( $assoc_args, 'retry', 0 );
+			$retry_delay = (int) Utils\get_flag_value( $assoc_args, 'retry-delay', 5 );
 
-			if ( 404 === (int) $response->status_code ) {
-				WP_CLI::error( 'Release not found. Double-check locale or version.' );
-			} elseif ( 20 !== (int) substr( $response->status_code, 0, 2 ) ) {
-				WP_CLI::error( "Couldn't access download URL (HTTP code {$response->status_code})." );
-			}
+			do {
+				$response       = Utils\http_request( 'GET', $download_url, null, $headers, $options );
+				$status_code    = (int) $response->status_code;
+				$is_response_ok = $status_code >= 200 && $status_code < 300;
+
+				// Exit the loop if the response is OK or the status code is 404.
+				if ( $is_response_ok || 404 === $status_code ) {
+					$retry = 0;
+				}
+
+				if ( 404 === $status_code ) {
+					WP_CLI::error( 'Release not found. Double-check locale or version.' );
+				} elseif ( ! $is_response_ok ) {
+					WP_CLI::warning( "Couldn't access download URL (HTTP code {$response->status_code})." );
+					if ( 0 < $retry ) {
+						WP_CLI::log( "Retrying in {$retry_delay} seconds..." );
+						sleep( $retry_delay );
+					}
+				}
+			} while ( 0 < $retry-- && ! $is_response_ok );
 
 			if ( 'nightly' !== $version ) {
 				unset( $options['filename'] );
-				$md5_response = Utils\http_request( 'GET', $download_url . '.md5', null, [], $options );
-				if ( $md5_response->status_code >= 200 && $md5_response->status_code < 300 ) {
-					$md5_file = md5_file( $temp );
+				$retry = (int) Utils\get_flag_value( $assoc_args, 'retry', 0 );
 
-					if ( $md5_file === $md5_response->body ) {
-						WP_CLI::log( 'md5 hash verified: ' . $md5_file );
+				do {
+					$md5_response = Utils\http_request( 'GET', $download_url . '.md5', null, [], $options );
```

Comment:
Just discussed the retries with @schlessera and the Requests library does provide a way to hook options in curl - https://requests.ryanmccue.info/docs/hooks.html#:~:text=Requests%5Crequest_multiple().-,curl.before_request,-Set%20cURL%20options
#259

Not sure though if we should add it by default in `Utils\http_request()`. Thoughts @schlessera?